backend/googlecloudstorage/googlecloudstorage.go

```diff
@@ -1063,9 +1113,21 @@ func (o *Object) Update(ctx context.Context, in io.Reader, src fs.ObjectInfo, op
 		ContentType: fs.MimeType(ctx, src),
 		Metadata:    metadataFromModTime(modTime),
 	}
+
+	buf := o.fs.pool.Get()
+
+	// Google library expects buffer of 0 len with enough capacity.
+	buf = buf[:0]
```

Comment:
This is a good practice I think that this is handled by pool impl, if it isn't it should be.